arvo/jael/tasks.md

```diff
@@ -0,0 +1,238 @@
+
+## Examples
+
+See the [%public-keys and %nuke](@/docs/arvo/jael/examples.md#public-keys-and-nuke) section of the Examples document for an example of using `%nuke` to cancel a `%public-keys` subscription. See the thread in the [%private-keys](@/docs/arvo/jael/examples.md#private-keys) example for cancelling a `%private-keys` subscription.
+
+# `%private-keys`
+
+```hoon
+[%private-keys ~]
+```
+
+Subscribe to private key updates.
+
+An agent or thread can subscribe to be notified of private key updates for the local ship. The subscription will continue until Jael receives a [%nuke](#nuke) `task` to cancel it.
+
+## Returns
+
+Jael responds to a `%private-keys` `task` with a `%private-keys` `gift` which looks like:
+
+```hoon
+[%private-keys =life vein=(map life ring)]
+```
+
+The `life` is the current life of the ship and the `vein` `map` contains the private key for each life up to (and including) the current one. Upon first subscribing, Jael will immediately respond with a `%private-keys` `gift`. Then, whenever the ship's private keys are changed, it'll send a new and updated `%private-keys` `gift`.
+
+## Example
+
+See the [%private-keys](@/docs/arvo/jael/examples.md#private-keys) section of the Examples document for a practical example.
+
+# `%public-keys`
+
+```hoon
+[%public-keys ships=(set ship)]
+```
+
+Subscribe to public key (and related) updates from Jael.
+
+An agent or thread can subscribe to be notified of public key updates, sponsorship changes and continuity breaches for the `set` of `ship`s specified in the `ships` field. The subscription will continue until Jael receives a [%nuke](#nuke) `task` to cancel it.
+
+## Returns
+
+Jael responds to a `%public-keys` `task` with `%public-keys` `gift`s which look like:
+
+```hoon
+[%public-keys =public-keys-result]
+```
+
+The [$public-keys-result](@/docs/arvo/jael/data-types.md#public-keys-result) contains whatever changes have occurred.
+
+Upon subscription, Jael will immeditely respond with a `%public-keys` `gift` containing a `%full` `public-keys-result` with the public key for each `life` up until the current one for each `ship` specified in the original `task`. After than, Jael will send a `%public-keys` `gift` with either a `%diff` or `%breach` `public-keys-result` each time a change occurs for any of the `ship`s to which you're subscribed.
+
+## Example
+
+See the [%public-keys and %nuke](@/docs/arvo/jael/examples.md#public-keys-and-nuke) section of the Examples document for a practical example.
+
+# `%rekey`
+
+```hoon
+[%rekey =life =ring]
+```
+
+Update private keys.
+
+This is what is send to Eyre by `%hood` implicitly when you run `|rekey` after rekeying via Bridge (or via `|cycle-moon-keys` on your planet in the case of a moon). It will update your `life` (key revision number) and private keys. The `life` field is the new `life` (typically an increment of the current `life`) and the `ring` is a private key `@`.
```

Comment:
```suggestion
This is what is sent to Jael by `%hood` when you run `|rekey`, as you must after setting new Azimuth keys. (In the case of a moon, `|cycle-moon-keys` sends this for you). It will update your `life` (key revision number) and private keys. The `life` field is the new `life` (typically an increment of the current `life`) and the `ring` is a private key `@`.
```

Or some similar change that's less awkwardly worded.